packages/manager/apps/sign-up/src/incomplete/routing.js

```diff
@@ -0,0 +1,33 @@
+import component from './component';
+
+import { NIC_STATUS_ENUM } from '../constants';
+
+import hourglassImg from './assets/img/hourglass.png';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('sign-up.incomplete', {
+    url: 'incomplete',
+    views: {
+      '@': {
+        component: component.name,
+      },
+    },
+    redirectTo: (transition) => {
+      const mePromise = transition.injector().getAsync('me');
+      return mePromise.then(({ state }) => {
+        if (state !== NIC_STATUS_ENUM.INCOMPLETE) {
+          return transition.router.stateService.target('sign-up', {}, {
```

Comment:
is there a simpler way of doing that ? like ```transition.redirect(...)``` ?

Reply:
I don't know. The difficulty here was to find a way to redirect without changing the location. This is the only way I found for doing it

Reply:
I think you can pass a `lcoation: false` if you use a `$state.go`, this will keep the current location